python/pyspark/sql/context.py

```diff
@@ -714,14 +711,13 @@ def __init__(self, sparkContext: SparkContext, jhiveContext: Optional[JavaObject
             + "SparkSession.builder.enableHiveSupport().getOrCreate() instead.",
             FutureWarning,
         )
+        static_conf = {}
         if jhiveContext is None:
-            sparkContext._conf.set(  # type: ignore[attr-defined]
-                "spark.sql.catalogImplementation", "hive"
-            )
-            sparkSession = SparkSession.builder._sparkContext(sparkContext).getOrCreate()
-        else:
-            sparkSession = SparkSession(sparkContext, jhiveContext.sparkSession())
-        SQLContext.__init__(self, sparkContext, sparkSession, jhiveContext)
+            static_conf = {"spark.sql.catalogImplementation": "in-memory"}
```

Comment:
Shoot. I pushed a wrong change in the last minute - I was manually testing if this value is correctly set or not. I has to be `hive`. This uses an existing session that has `hive` by default so it doesn't affect the change but would impact shell if users use `HiveContext` directly.

Sorry this was my mistake. I will make a quick followup.

Reply:
https://github.com/apache/spark/pull/35517